### PR TITLE
Add snakeobjects

### PR DESCRIPTION
**Pull request by yamrom**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by yamrom** [comment]
All checks have passed. How to ask for approving review?

File: recipes/snakeobjects/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package:
+  name: snakeobjects
+  version: "3.0"
+
+source:
+  #url: https://github.com/iossifovlab/{{ name|lower }}/archive/{{ version }}.tar.gz
```
Comment (simonbray) [inline recipes/snakeobjects/meta.yaml:6]: Can you remove commented out lines please?

File: recipes/snakeobjects/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - python
+    - snakemake
+
+test:
+  source_files:
+    - tests
+  requires:
+    - pytest
+    - pytest-cov
+    - snakemake
```
Comment (simonbray) [inline recipes/snakeobjects/meta.yaml:40]:
```suggestion
  commands:
    - sobjects --help
  imports:
    - ...
```

Comment (simonbray) [reply recipes/snakeobjects/meta.yaml:40]: in general we don't include run an entire test suite as part of bioconda tests

File: recipes/snakeobjects/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+about:
+  home: https://github.com/iossifovlab/snakeobjects
+  summary: Our pipeline infrastructure.
```
Comment (simonbray) [inline recipes/snakeobjects/meta.yaml:44]: Can you improve this summary please?

File: recipes/snakeobjects/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  license: MIT
+  license_file: LICENSE
```
Comment (simonbray) [inline recipes/snakeobjects/meta.yaml:46]: Can you add yourself as a maintainer to this package?

File: recipes/snakeobjects/run_test.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
Comment (simonbray) [inline recipes/snakeobjects/run_test.sh:1]: this file can be removed

File: recipes/snakeobjects/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package:
+  name: snakeobjects
+  {% set version = "3.0" %}
```
Comment (simonbray) [inline recipes/snakeobjects/meta.yaml:3]:
```suggestion
{% set version = "3.0" %}
package:
  name: snakeobjects
  version: {{ version }}
```

File: recipes/snakeobjects/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+about:
+  home: https://github.com/iossifovlab/snakeobjects
+  summary: Snakeobjects, an object-oriented workflow management system based on snakemake
+  license: MIT
+  license_file: LICENSE
+
+extra:
+  maintainers:
+   - Boris Yamrom
```
Comment (simonbray) [inline recipes/snakeobjects/meta.yaml:48]:
```suggestion
   - yamrom
```
better to use a github username, then you will get a notification if any changes are made to the recipe

**Comment by yamrom** [comment]
Please review and merge

**Comment by simonbray** [comment]
@yamrom can you remove all comments please?

**Comment by yamrom** [comment]
removed all comments

**Review by simonbray — APPROVED** [review]

**Comment by simonbray** [comment]
@bioconda-bot please merge

**Pull request merged**